src/btreemap/allocator.rs

```diff
@@ -94,6 +95,18 @@ impl<M: Memory> Allocator<M> {
         allocator
     }
 
+    /// Deallocate all allocated chunks.
+    pub fn clear(&mut self) {
```

Comment:
There is duplicate logic between this function and `fn new()`. Maybe we can merge them together?

```
    pub fn new(memory: M, addr: Address, allocation_size: Bytes) -> Self {
        let mut allocator = Self {
            header_addr: addr,
            allocation_size,
            num_allocated_chunks: 0,
            free_list_head: NULL, // Will be set in `allocator.clear()` below.
            memory,
        };

        allocator.clear();
        allocator
    }

```